viamonvif/service.go

```diff
@@ -175,21 +178,43 @@ func (dis *rtspDiscovery) DoCommand(ctx context.Context, command map[string]inte
 		if err != nil {
 			return nil, err
 		}
+
+		var snapshotErr, rtspErr error
+
+		// Attempt to fetch the image from the snapshot URI
 		dis.rtspToSnapshotURIsMu.Lock()
 		snapshotURI, found := dis.rtspToSnapshotURIs[previewReq.rtspURL]
 		dis.rtspToSnapshotURIsMu.Unlock()
-		if !found {
-			return nil, fmt.Errorf("snapshot URI not found for %s", previewReq.rtspURL)
+
+		if found {
+			dis.logger.Debugf("attempting to fetch image from snapshot URI: %s", snapshotURI)
+			dataURL, err := downloadPreviewImage(ctx, dis.logger, snapshotURI)
+			if err == nil {
+				dis.logger.Debugf("Successfully fetched image from snapshot URI: %s", snapshotURI)
+				return map[string]interface{}{
+					"preview": dataURL,
+				}, nil
+			}
+			dis.logger.Warnf("failed to fetch image from snapshot URI: %s, error: %v", snapshotURI, err)
+			snapshotErr = fmt.Errorf("snapshot error: %w", err)
+		} else {
+			dis.logger.Warnf("snapshot URI not found for RTSP URL: %s", previewReq.rtspURL)
+			snapshotErr = errors.New("snapshot URI not found")
```

Comment:
should we add the url to the error itself too

Reply:
Added